Framework/Algorithms/src/ChangePulsetime.cpp

```diff
@@ -38,41 +36,27 @@ void ChangePulsetime::init() {
 /** Execute the algorithm.
  */
 void ChangePulsetime::exec() {
-  EventWorkspace_const_sptr in_ws = getProperty("InputWorkspace");
+  EventWorkspace_sptr in_ws;
+  SpectrumIndexSet indexSet(0);
+
+  std::tie(in_ws, indexSet) = std::tuple<EventWorkspace_sptr, SpectrumIndexSet>(
+      getProperty("InputWorkspace"));
+
   EventWorkspace_sptr out_ws = getProperty("OutputWorkspace");
   if (!out_ws) {
     out_ws = in_ws->clone();
   }
 
-  // Either use the given list or use all spectra
-  std::vector<int> workspaceIndices = getProperty("WorkspaceIndexList");
-  int64_t num_to_do = static_cast<int64_t>(workspaceIndices.size());
-  bool doAll = false;
-  if (workspaceIndices.empty()) {
-    doAll = true;
-    num_to_do = in_ws->getNumberHistograms();
-  }
-
   double timeOffset = getProperty("TimeOffset");
+  Progress prog(this, 0.0, 1.0, indexSet.size());
 
-  Progress prog(this, 0.0, 1.0, num_to_do);
-  PARALLEL_FOR_NO_WSP_CHECK()
```

Comment:
I need to change the loop I think. OPENMP requires for in the canonical form unless I do something like [this](https://stackoverflow.com/questions/17848521/using-openmp-with-c11-range-based-for-loops)